stan/math/prim/scal/fun/boost_policy.hpp

```diff
@@ -19,7 +19,8 @@ namespace math {
 typedef boost::math::policies::policy<
     boost::math::policies::overflow_error<
         boost::math::policies::errno_on_error>,
-    boost::math::policies::pole_error<boost::math::policies::errno_on_error> >
+    boost::math::policies::pole_error<boost::math::policies::errno_on_error>,
+    boost::math::policies::promote_double<false> >
```

Comment:
@wds15, why is the default policy changing? This is not listed in the side-effects.

Reply:
I will add it to the side-effects (I think I had it before, but probably dropped to note this in the iterations). The thing is: boost is by default configured to run with long double precision internally when you use double arguments. This makes the lgamma boost function more accurate (they say so), but also extremely slow. From what I can tell, we don't need this extra accuracy; at least the tests come out basically the same with the boost and the std version.

Reply:
The lgamma is used in a bunch of special functions and gradients.  Some of them use iterative algorithms and are very sensitive to accumulated error.  If we're dropping the accuracy on lgamma those will be affected and most of them don't have the test coverage required to detect the issue.  The result is faster evaluation of the log density but a dropping acceptance rate and sampling from the wrong density so it's not straightforward to diagnose.  Could we meet the goals here without dropping accuracy?

Reply:
We changed in the past the boost lgamma with high precision to the std lgamma for a few versions now and no one noticed problems. Now we will use lgamma_r from c on macOS and Linux. On windows we use again the boost lgamma at lower precision. The lower precision will affect the digamma function on all platforms. From the negative binomial example above I got the same results at twice the speed. Sounds ok to me.